### PR TITLE
[SYCL][Windows] Enable building lld by default on Windows

### PR DESCRIPTION
**Pull request by npmiller**
The Windows-Clang CMake module uses `lld-link` by default, so
having lld built on windows is helpful for people wanting to build SYCL
applications using CMake.

Related to: https://github.com/intel/llvm/issues/6026

### CODE REVIEW

File: sycl/CMakeLists.txt [hunk 1]
```diff
@@ -383,6 +383,12 @@ if(OpenCL_INSTALL_KHRONOS_ICD_LOADER AND TARGET OpenCL-ICD)
   list(APPEND SYCL_TOOLCHAIN_DEPLOY_COMPONENTS OpenCL-ICD)
 endif()
 
+# On Windows or for HIP AMD lld is also needed
+if(WIN32 OR ("hip" IN_LIST SYCL_ENABLE_PLUGINS AND "${SYCL_BUILD_PI_HIP_PLATFORM}" STREQUAL "AMD"))
```
Comment (bader) [inline sycl/CMakeLists.txt:387]: Can we simplify the condition here to following logic:
 - if lld is added to the list of enabled project then install it as part of sycl toolchain.
?

This should remove the duplication of the cmake configuration logic.

Comment (npmiller) [reply sycl/CMakeLists.txt]: Good point, simplified the condition as suggested and added an error for HIP if `lld` is missing

File: sycl/CMakeLists.txt [hunk 2]
```diff
@@ -383,6 +383,12 @@ if(OpenCL_INSTALL_KHRONOS_ICD_LOADER AND TARGET OpenCL-ICD)
   list(APPEND SYCL_TOOLCHAIN_DEPLOY_COMPONENTS OpenCL-ICD)
 endif()
 
+# Build and install lld as part of the sycl-toolchain if available
+if(TARGET lld)
```
Comment (bader) [inline sycl/CMakeLists.txt:387]: I hope it works.
I thought about examining LLVM_ENABLE_PROJECTS value, but assuming that lld's project cmake files are parsed before this cmake file, this should work as well.

**Review by bader — APPROVED** [review]

File: sycl/CMakeLists.txt (hunk 2 above)
Comment (npmiller) [reply sycl/CMakeLists.txt]: Good point, I was under the incorrect assumptions that the target would need to already be defined for `add_dependencies` to work anyway, but it looks like `add_dependencies` still works even if the target is defined later, so I've replaced the conditions with your initial suggestion.

**Review by bader — APPROVED** [review]

**Review by aelovikov-intel — APPROVED** [review]

**Pull request merged**